pkg/controller/signup.go

```diff
@@ -46,6 +47,51 @@ func (s *Signup) PostHandler(ctx *gin.Context) {
 	ctx.Writer.WriteHeaderNow()
 }
 
+// UpdateVerificationHandler stars the verification process and updates a usersignup resource
+func (s *Signup) UpdateVerificationHandler(ctx *gin.Context) {
+	userID := ctx.GetString(context.SubKey)
+	signup, err := s.signupService.GetUserSignup(userID)
+	if err != nil {
+		if errors2.IsNotFound(err) {
+			log.Error(ctx, err, "usersignup not found")
+			errors.AbortWithError(ctx, http.StatusNotFound, err, "usersignup not found")
+		}
+		log.Error(ctx, err, "error retreiving usersignup")
+		errors.AbortWithError(ctx, http.StatusInternalServerError, err, fmt.Sprintf("error retreiving usersignup: %s", userID))
+		return
+	}
+
+	// Read the Body content
+	var bodyBytes []byte
+	if ctx.Request.Body != nil {
+		bodyBytes, _ = ioutil.ReadAll(ctx.Request.Body)
+	}
+
```

Comment:
```suggestion
	defer func() {
		if err := res.Body.Close(); err != nil {
			log.Error(nil, err, "failed to close response after reading")
		}
	}
```

Reply:
you also need wrap the call `res.Body.Close()` in a deferred func

Reply:
fixed here: 7908cfab4b879d2a292e2d8dafb882a6bbe3b967 and here: https://github.com/codeready-toolchain/registration-service/blob/7908cfab4b879d2a292e2d8dafb882a6bbe3b967/pkg/controller/signup.go#L70-L85